src/commands/lint.ts

```diff
@@ -17,117 +17,182 @@
+  }
+}`);
+      process.exitCode = 1;
+      return;
     }
 
-    // Default to false if input files are provided, otherwise default to true
-    let followDependencies = !options['input'];
-    if (options['follow-dependencies']) {
-      followDependencies = true;
-    } else if (options['no-follow-dependencies']) {
-      followDependencies = false;
+    const rules = lintLib.registry.getRules(ruleCodes || lintOptions.rules);
+    const filter = new WarningFilter({
+      warningCodesToIgnore: new Set(lintOptions.ignoreWarnings || []),
+      minimumSeverity: Severity.WARNING
+    });
+
+
+    const analyzer = new Analyzer({
+      urlLoader: new FSUrlLoader(config.root),
+      urlResolver: new PackageUrlResolver(),
+    });
+
+    const linter = new lintLib.Linter(rules, analyzer);
+
+    const warnings = await this._lint(linter, options.input);
+
+    const filtered = warnings.filter((w) => !filter.shouldIgnore(w));
+
+    const printer = new WarningPrinter(
+        process.stdout, {analyzer: analyzer, verbosity: 'full', color: true});
+    await printer.printWarnings(filtered);
+
+    process.exitCode = this._getExitCode(filtered);
+  }
+
+  extraUsageGroups(config: ProjectConfig): UsageGroup[] {
+    const lintLib: typeof lintLibTypeOnly = require('polymer-linter');
+    const chalk: typeof chalkTypeOnly = require('chalk');
+    this._loadPlugins(config);
+    let collectionsDocs = [];
+    for (const collection of lintLib.registry.allRuleCollections) {
+      collectionsDocs.push(`  ${chalk.bold(collection.code)}: ${this._indent(
+          collection.description)}`);
+    }
+    let rulesDocs = [];
+    for (const rule of lintLib.registry.allRules) {
+      rulesDocs.push(
+          `  ${chalk.bold(rule.code)}: ${this._indent(rule.description)}`);
     }
+    return [
+      {
+        header: 'Lint Rule Collections',
+        content: collectionsDocs.join('\n\n'),
+        raw: true
+      },
+      {header: 'Lint Rules', content: rulesDocs.join('\n\n'), raw: true}
+    ];
+  }
 
-    return polylint
-        .runWithOptions({
-          input: lintFiles,
-          root: config.root,
-          // TODO: read into config
-          bowerdir: 'bower_components',
-          policy: options['policy'],
-          'config-file': options['config-file'],
-          'config-field': options['config-field'],
-          // NOTE: `no-recursion` has the opposite behavior of
-          // `follow-dependencies`
-          'no-recursion': !followDependencies,
+  private _indent(description: string) {
+    return description.split('\n')
+        .map((line, idx) => {
+          if (idx === 0) {
+            return line;
+          }
+          if (line.length === 0) {
+            return line;
+          }
+          return '      ' + line;
         })
-        .then(() => undefined);
+        .join('\n');
+  }
+
+  private _loadPlugins(_config: ProjectConfig) {
+    // TODO(rictic): implement.
+  }
+
+  private async _lint(
```

Comment:
This may be personal preference, but I prefer to keep the action of a main function like `run()` inlined instead of abstracted. The contents of this helper function are small enough that the abstraction doesn't meaningfully simplify the main function it was taken from (and caused me to jump back and forth while reading `run()` to understand what it was actually doing).